engine/lib/metadata.php

```diff
@@ -98,78 +98,18 @@ function elgg_delete_metadata(array $options) {
  */
 
-/**
- * Returns entities based upon metadata.  Also accepts all
- * options available to elgg_get_entities().  Supports
- * the singular option shortcut.
- *
- * @note Using metadata_names and metadata_values results in a
- * "names IN (...) AND values IN (...)" clause.  This is subtly
- * differently than default multiple metadata_name_value_pairs, which use
- * "(name = value) AND (name = value)" clauses.
- *
- * When in doubt, use name_value_pairs.
- *
- * To ask for entities that do not have a metadata value, use a custom
- * where clause like this:
- *
- * 	$options['wheres'][] = "NOT EXISTS (
- *			SELECT 1 FROM {$dbprefix}metadata md
- *			WHERE md.entity_guid = e.guid
- *				AND md.name = $name
- *				AND md.value = $value)";
- *
- * @see elgg_get_entities
- *
- * @param array $options Array in format:
- *
- * 	metadata_names => null|ARR metadata names
- *
- * 	metadata_values => null|ARR metadata values
- *
- * 	metadata_name_value_pairs => null|ARR (
- *                                         name => 'name',
- *                                         value => 'value',
- *                                         'operand' => '=',
- *                                         'case_sensitive' => true
- *                                        )
- *                               Currently if multiple values are sent via
- *                               an array (value => array('value1', 'value2')
- *                               the pair's operand will be forced to "IN".
- *                               If passing "IN" as the operand and a string as the value,
- *                               the value must be a properly quoted and escaped string.
- *
- * 	metadata_name_value_pairs_operator => null|STR The operator to use for combining
- *                                        (name = value) OPERATOR (name = value); default AND
- *
- * 	metadata_case_sensitive => BOOL Overall Case sensitive
- *
- *  order_by_metadata => null|ARR array(
- *                                      'name' => 'metadata_text1',
- *                                      'direction' => ASC|DESC,
- *                                      'as' => text|integer
- *                                     )
- *                                Also supports array('name' => 'metadata_text1')
- *
- * @return \ElggEntity[]|mixed If count, int. If not count, array. false on errors.
- * @since 1.7.0
- */
-function elgg_get_entities_from_metadata(array $options = []) {
-	return _elgg_services()->metadataTable->getEntities($options);
-}
-
 /**
  * Returns a list of entities filtered by provided metadata.
  *
- * @see elgg_get_entities_from_metadata
+ * @see elgg_get_entities
  *
  * @param array $options Options array
  *
  * @return array
  * @since 1.7.0
  */
 function elgg_list_entities_from_metadata($options) {
-	return elgg_list_entities($options, 'elgg_get_entities_from_metadata');
+	return elgg_list_entities($options, 'elgg_get_entities');
```

Comment:
`elgg_list_entities` defaults to the `elgg_get_entities` getter, so could go away

Reply:
Yes, need to deprecate all `elgg_list_` functions as well